src/main/java/com/pi4j/crowpi/components/SoundSensorComponent.java

```diff
@@ -0,0 +1,175 @@
+package com.pi4j.crowpi.components;
+
+import com.pi4j.context.Context;
+import com.pi4j.crowpi.components.events.DigitalEventProvider;
+import com.pi4j.crowpi.components.events.EventListener;
+import com.pi4j.crowpi.components.events.SimpleEventHandler;
+import com.pi4j.io.gpio.digital.DigitalInput;
+import com.pi4j.io.gpio.digital.DigitalInputConfig;
+import com.pi4j.io.gpio.digital.DigitalState;
+import com.pi4j.io.gpio.digital.PullResistance;
+
+/**
+ * Implementation of the CrowPi sound sensor using GPIO with Pi4J
+ */
+public class SoundSensorComponent extends Component implements DigitalEventProvider<SoundSensorComponent.SoundState> {
+    /**
+     * Pi4J digital input instance used by this component
+     */
+    protected final DigitalInput digitalInput;
+    /**
+     * Default Pin of sound sensor
+     */
+    protected static final int DEFAULT_PIN = 24;
+    /**
+     * Debounce of input in microseconds
+     */
+    protected static final long DEFAULT_DEBOUNCE = 1000;
+
+    /**
+     * Handler for simple event when a noise is recognized
+     */
+    private SimpleEventHandler onNoise;
+    /**
+     * Handler for simple event when it's silent again
+     */
+    private SimpleEventHandler onSilence;
+
+    /**
+     * Creates a new sound sensor component using the default setup.
+     *
+     * @param pi4j Pi4J context
+     */
+    public SoundSensorComponent(Context pi4j) {
+        this(pi4j, DEFAULT_PIN, DEFAULT_DEBOUNCE);
+    }
+
+    /**
+     * Create sound sensor component with custom parameters
+     *
+     * @param pi4j     Pi4J context
+     * @param address  GPIO Pin of Raspberry
+     * @param debounce Time to debounce the sound sensor in microseconds
+     */
+    public SoundSensorComponent(Context pi4j, int address, long debounce) {
+        this.digitalInput = pi4j.create(buildDigitalInputConfig(pi4j, address, debounce));
+        this.addListener(this::dispatchSimpleEvents);
+    }
+
+    /**
+     * Sets or disables the handler for the onNoise event.
+     * This event gets triggered whenever the sensor registers noise.
+     * Only a single event handler can be registered at once.
+     *
+     * @param handler Event handler to call or null to disable
+     */
+    public void onNoise(SimpleEventHandler handler) {
+        this.onNoise = handler;
+    }
+
+    /**
+     * Reads the current sensor state
+     *
+     * @return Returns true if there is a noise currently registered by the sensor
+     */
+    public boolean isNoisy() {
+        return digitalInput.isLow();
```

Comment:
I would use `getState() == SoundState.NOISE` here to use the mapped value instead of determining directly based on the digital input. Our existing components do so too, so it would make sense to keep this approach consistent.

Also I think this comparison is wrong? According to `mapDigitalState` a value of `LOW` is supposed to be silence but this method says the opposite?